src/super_gradients/training/datasets/depth_estimation_datasets/redweb_dataset.py

```diff
@@ -0,0 +1,57 @@
+import os
+
+from typing import List
+
+from PIL import Image
+
+from super_gradients.training.datasets.depth_estimation_datasets import AbstractDepthEstimationDataset
+from super_gradients.training.transforms.depth_estimation import AbstractDepthEstimationTransform
+from super_gradients.training.samples import DepthEstimationSample
+from super_gradients.common.object_names import Datasets
+from super_gradients.common.registry.registry import register_dataset
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from super_gradients.common.factories.transforms_factory import TransformsFactory
+
+
+@register_dataset(Datasets.REDWEB_DEPTH_ESTIMATION_DATASET)
+class ReDWebDepthEstimationDataset(AbstractDepthEstimationDataset):
+    """
+    Dataset class for training depth estimation models using ReDWeb V1 dataset.
+
+    Link to download dataset: https://sites.google.com/site/redwebcvpr18/
+    Paper: https://openaccess.thecvf.com/content_cvpr_2018/papers/Xian_Monocular_Relative_Depth_CVPR_2018_paper.pdf
+
+    The ReDWeb V1 dataset consists of 3.6K RGB-RD images, covering both indoor and outdoor scenes.
+    """
+
+    @resolve_param("transforms", factory=TransformsFactory())
+    def __init__(
+        self,
+        data_dir: str,
+        transforms: List[AbstractDepthEstimationTransform] = [],
+        images_dir: str = "Imgs",
+        targets_dir: str = "RDs",
+        image_extension: str = "jpg",
+        target_extension: str = "png",
+    ):
+        """ """
+        super().__init__(transforms=transforms)
+        self.data_dir = data_dir
+        self.images_dir = images_dir
+        self.targets_dir = targets_dir
+        self.image_extension = image_extension
+        self.target_extension = target_extension
+
+        self.pair_names = [name.split(".")[0] for name in os.listdir(os.path.join(self.data_dir, self.targets_dir))]
```

Comment:
A few issues I see here:
1) `sorted(os.listdir(...))` Just to stay safe. listdir is not guaranteed to return file names in same order.
2) We are not validating that os.listdir returns all images of target_extension.
3) We are not validating that present images has the same filenames (set are equal) as targets.

So a better approach would be to:
- get all files in images dir 
- all files in targets dir
- filter both of them according to `image_extension` and `target_extension`
- check that two sets are equal

Reply:
In the case of a ReDWeb dataset, I haven't seen any issues with file names, so an approach with storing only names doesn't require two additional lists with file paths and any kind of sorting. 

Do we foresee the case that users will somehow change the structure of the dataset offline?

Reply:
> Do we foresee the case that users will somehow change the structure of the dataset offline?
Yes, we don't trust no one

- User can forget to put a correct dataset class (E.g instead say Hypersim put ReDWeb)
- User can have corrupted dataset or just copy some images by accident

The idea is to fail as early as possible.
If the dataset is not consistent and we are tryin to read non-existent image, with current design user will get an exception from the DataLoader at most which won't be telling much (Cannot read image bla-bla-bla.png)
When we check consistency in class c'tor - we can provide a more _meaningful_ message _earlier_.